### DISCUSSION TITLE
`BitButton` should trigger form onsubmit when inside a form

### DISCUSSION THREAD

**Discussion by linkdotnet** [bitfoundation/bitplatform#2866] [Ideas]
I am not sure if this is really feasible and I do understand that the idea is arguable at best. But anyway:

Any button, which doesn't define explicitly the `type="button"` inside a form, will be of `type="submit"`. 

> The missing value default and invalid value default are the Submit Button state.

Specification can be found here: https://html.spec.whatwg.org/multipage/form-elements.html#attr-button-type

That means if we have a component like this:
```razor
<EditForm Model="@MyModel" OnValidSubmit="HandleSubmit">
    <BitButton>Submit</BitButton>
</EditForm>

@code {
    private void HandleSubmit()
    {
        // Something to be invoked here
    }
}
```

Clicking on the button would invoke `HandleSubmit`. Only explicitly setting it to `ButtonType="BitButtonType.Submit"`. 
Now I do understand that the default is `ButtonType.Button` which translates to `type="button"`. The confusion comes from the point that if you would do that in html (without specifing the type) the form would be submitted.

One possible solution would be to make `ButtonType` optional aka nullable and check whether or not a `EditContext` is available.

**Comment by linkdotnet** [comment]
If wished I can try to create a Pull-Request. The described solution above would be a breaking change. As you released `v2` not so long ago, that might be not ideal.

**Comment by msynk** [comment]
thanks for pointing out the issue. I agree with your solution, it's feasible and I think it works fine.
let us discuss its impact on the current implementations and the effect of the breaking change.
although, submitting a PR to fix this would be great.

**Reply by linkdotnet** [reply]
Cool. Thanks for the super swift answer! Much appreciated.

**Reply by linkdotnet** [reply]
PR: https://github.com/bitfoundation/bitplatform/pull/2872
